src/adam_bashforth_adaptive.jl

```diff
@@ -0,0 +1,371 @@
+    # n x order over course of algorithm, as we use n steps, where n is
+    #unknown
+    Tdy=typeof(F(t[1], y[1]))
+    c,g,b,Φstar, Φ = intialize_coef_arrays(order+3,Tdy)
+
+    #Array which holds norms of LE(n+1), the local error or truncation error, which will
+    #will used in step size selection. normLE[1] corresponds to norm of LE(n+1)_{k-1},
+    # normLE[2] to norm of LE(n+1)_{k},  normLE[3] to norm of  LE(n+1)_{k+1},
+    #and normLE[4] to norm of  LE(n+1)_{k+2},
+    normLE = [0.0 0.0 0.0 0.0]
+
+    ##Note that due to Julia starting indexing at 1, k = steporder + 1 in what follows
+    #start at n=1, the first step
+    n = 1
+    steporder=1
+    wait=0
+    while abs(t[n] - tfinal) > 0 && sign(tfinal-t[n])==tdir #&&  minstep < abs(h[n])
+        # Need to run the first several steps at reduced order, and need to maintain
+        # n>steporder invariant in order to perform adaptive step selection of the
+        # 'order' = steporder, for it requires the last 'order' number of points
+        if n<order
+            steporder = n==1 ? 1 : min(n-1, order)
+        end
+
+        #allocate space for t[n+1], y[n+1], dy[n+1], h[n+1]
+        t,y,dy, h = increment_length(t,y,dy,h,n)
+
+        successful_step = false
+        while !successful_step
+          #calculation (n+1)-th time position from n-th step size
+          t[n+1]=t[n]+h[n]
+          #####################################################################
+          #Calculating explicit approximatiion to y[n+1]
+          #####################################################################
+          #Current derivative: set dy[n] = F(t[n], y[n])
+          dy[n]=F(t[n], y[n])
+
+          #calculation of c_{j,q} coefficients for j=0,...,k-1; q = 0,...,k-1-(j-1)
+          g = g_coefs(g,c,h,t,n,min(n,steporder+3))
+
+          #calculation of b_j(n), Φstar_j(n), and Φ_j(n) coefficients from recurence relations
+          b, Φ, Φstar=Φ_and_Φstar_coefs(b, Φ, Φstar,dy[n],h,t,n,1:min(max(n-1,1),steporder+1))
+
+          ##Calculate next value, y[n+1], with order = k = steporder + 1. See
+          #equation (5.5) in Hairer et al, Vol 1, where we note that here
+          # y[n+1] = y_{k}(t_{n+1})
+          y[n+1] = y[n]
+          for j= 1:steporder
+            y[n+1] += h[n]*g[j]* Φstar[2,j]
+          end
+
+          dy[n+1] = F(t[n+1],y[n+1])
+          #NOTE:fixed step size explicit solver tested and gives correct result
+          ######################################################################
+          #CE Loop for implicit solver
+          ######################################################################
+          pred = y[n+1]
+          # Set CE_loops to be how many correction-evaulation cycles
+          # desired. As least one loop is necessary, for the sake of the
+          # adaptive step selection
+          CE_step = 1
+          i = 0
+          if i<CE_step && n>=steporder+1
```

Comment:
Yes, it should be one. I switched to a `if` statement when I was testing with just one CE-loop, for some silly reason I can't recall. I switched back to a while loop